artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContext.java

```diff
@@ -326,6 +325,37 @@ private void actualDelivery(Delivery delivery, Receiver receiver, ReadableBuffer
       }
    }
 
+   private DeliveryState determineDeliveryState(final Source source, final boolean useModified, final Exception e) {
+      Outcome defaultOutcome = getEffectiveDefaultOutcome(source);
+      if (e instanceof ActiveMQAddressFullException && useModified &&
+          (outcomeSupported(source, Modified.DESCRIPTOR_SYMBOL) || defaultOutcome instanceof Modified)) {
+         Modified modified = new Modified();
+         modified.setDeliveryFailed(true);
+         modified.setUndeliverableHere(false);
```

Comment:
I was too focused on the Accept and looking into its implications earlier that I missed this. I don't think undeliverable-here should be set, or else it should be configurable. There is nothing wrong with the message itself here, and the point of the related JIRA was you want the sender to be able to send the message again, but setting undeliverable-here specifically instructs them otherwise.

Reply:
`undeliverable-here` is set false.  only `delivery-failed` is set true.  I maybe missing your point.

Reply:
You aren't missing anything, I've somehow completely misread that the second time round, thinking it was being set true...no idea how, read start of one line and end of other? :)

I probably wouldn't bother setting it, but don't see harm in doing so.

Reply:
That should be fixed.